test/ecto/repo_test.exs

```diff
@@ -378,4 +381,60 @@ defmodule Ecto.RepoTest do
       assert TestRepo.preload(nil, []) == nil
     end
   end
+
+  test "Repo.load/2" do
+    # string fields
+    assert %MySchema{x: "abc"} =
+           TestRepo.load(MySchema, %{"x" => "abc"})
+
+    # atom fields
+    assert %MySchema{x: "abc"} =
+           TestRepo.load(MySchema, %{x: "abc"})
+
+    # keyword list
+    assert %MySchema{x: "abc"} =
+           TestRepo.load(MySchema, [x: "abc"])
+
+    # atom fields and values
+    assert %MySchema{x: "abc"} =
+           TestRepo.load(MySchema, {[:x], ["abc"]})
+
+    # string fields and values
+    assert %MySchema{x: "abc"} =
+           TestRepo.load(MySchema, {["x"], ["abc"]})
+
+    # default value
+    assert %MySchema{x: "abc", z: "z"} =
+           TestRepo.load(MySchema, %{x: "abc"})
+
+    # array field
+    assert %MySchema{array: ["one", "two"]} =
+           TestRepo.load(MySchema, %{array: ["one", "two"]})
+
+    # map field with atoms
+    assert %MySchema{map: %{color: "red"}} =
+           TestRepo.load(MySchema, %{map: %{color: "red"}})
+
+    # map field with strings
+    assert %MySchema{map: %{"color" => "red"}} =
+           TestRepo.load(MySchema, %{map: %{"color" => "red"}})
+
+    # nil
+    assert %MySchema{x: nil} =
+           TestRepo.load(MySchema, %{x: nil})
+
+    # invalid field is ignored
+    assert %MySchema{} =
+           TestRepo.load(MySchema, %{bad: "bad"})
+
+    # FIXME
+    # # invalid value
+    # assert_raise ArgumentError, "cannot load `0` as type :string for :x in schema Ecto.RepoTest.MySchema", fn ->
+    #   TestRepo.load(MySchema, %{x: 0})
+    # end
```

Comment:
just double-checking: it looks like a difference between `Ecto.Type.load` and `Ecto.Type.adapter_load` wrt fields of invalid type; the latter allowed loading `0` into a `string` field. Is that ok? If so, I'd need to remove this sentence from docs `If any of the values has invalid type, an error is raised.` or change it slightly.